### DISCUSSION TITLE
millenium mps-150 doesn't show up in midi input options

### DISCUSSION THREAD

**Discussion by karJac** [hydrogen-music/hydrogen#1750] [Q&A]
Hi y' all,

I'm trying to connect my edrums Millenium MPS-150 to Hydrogen on Windows 11 x64. Unfortunately it doesn't show up in midi input options.
![image](https://user-images.githubusercontent.com/98103985/232906568-063c957f-0702-48e2-9329-fdc2955e337c.png)

My system recognize my drumkit as e-Drum when I connect them through USB3.0
![image](https://user-images.githubusercontent.com/98103985/232906429-d1f6fff6-0d13-47b6-85bd-41cea0629674.png)

My input mode is set to Drumkit. I'm using hydrogen 1.2.0.

Do you have any ideas what's wrong?

Thank you in advance for your responses!
Karol

**Comment by theGreatWhiteShark** [comment]
Hey @karJac ,

Seems like Input combo box in your screenshot is disabled, isn't it? Does your kit works with other applications using PortMidi?

**Reply by karJac** [reply]
Thanks for response I will check it soon

**Reply by theGreatWhiteShark** [reply]
I just checked the code base. This combo box is disabled if either a) PortMidi does report no available devices or b) PortMidi was unable to retrieve information about any of the available devices.

Which one it is can be determined by Hydrogen's log file (accessible via the main menu). Could you open and attach it? In addition, which PortMidi version are you using?

**Reply by karJac** [reply]
Thank you a lot for that. The problem was of much simple nature -- you have to connect the device before running hydrogen. If you connect the device after hydrogen is already launched, it won't recognize it.

nevertheless still something weird is happening, the MIDI-in "diod" is lighted up all the time and system doesn't record hits on edrums.
![image](https://github.com/hydrogen-music/hydrogen/assets/98103985/c88c948b-afc0-4f10-b954-39a45d08c72c)

These are my settings:

![image](https://github.com/hydrogen-music/hydrogen/assets/98103985/f798406a-1c0a-41f7-b233-c22d83667f77)
![image](https://github.com/hydrogen-music/hydrogen/assets/98103985/deccf73c-eb6e-493f-8f2c-a1c1d4f38386)

And here are the logs:
[hydrogen.log](https://github.com/hydrogen-music/hydrogen/files/11472279/hydrogen.log)


Thank you a lot for helping me out and sorry for the long response time -- I was trying to fix it myself but now I gave up, the world of sound recording is something totally unknown to me :p

PS. Or maybe you know someone whom I could pay for setting it up on my laptop through remote display?

**Reply by theGreatWhiteShark** [reply]
> you have to connect the device before running hydrogen. If you connect the device after hydrogen is already launched, it won't recognize it.

What? That sounds quite strange. When using PortMidi the list of MIDI input devices should be rescanned every single time you open the Preferences. 

> nevertheless still something weird is happening, the MIDI-in "diod" is lighted up all the time and system doesn't record hits on edrums.

Yeah, that's a known bug of the 1.2.0 release. As a work around you can uncheck Preferences > MIDI System > "Discard MIDI messages after action has been triggered". Or you could use a pre-release of [version 1.2.1](https://ci.appveyor.com/api/buildjobs/0m3bawp3o97cmrek/artifacts/Hydrogen-1.2.0-win64.exe) where this issue is already fixed.

> PS. Or maybe you know someone whom I could pay for setting it up on my laptop through remote display?

Nope. But you are almost there.

**Reply by karJac** [reply]
The new version of hydrogen didn't solve problem with the MIDI "diod". I've check, that the  "Discard MIDI messages after action has been triggered" option was on. 
![image](https://github.com/hydrogen-music/hydrogen/assets/98103985/af6bcb36-1218-4464-b0fb-9000e7c2e440)

Here are the logs.
[hydrogen.log](https://github.com/hydrogen-music/hydrogen/files/11482077/hydrogen.log)

as always thank you for your time!

**Reply by theGreatWhiteShark** [reply]
> Here are the logs.
[hydrogen.log](https://github.com/hydrogen-music/hydrogen/files/11482077/hydrogen.log)

In the logs I currently just see floods of incoming MIDI messages of type [SYSTEM_EXCLUSIVE](https://en.wikipedia.org/wiki/MIDI#System_Exclusive_messages) . It's weird your drumkit sends them (especially since its [manual](https://images.static-thomann.de/pics/atg/atgdata/document/manual/c_360221_module_v3_r1_en_online.pdf) states that it is not even capable of sending them) but maybe it was designed to work with a special proprietary software. Those events are not - and can not - be handled by Hydrogen.

Could you increase the log level to "Info"? Then we hopefully will also see other MIDI messages send by your device

**Reply by theGreatWhiteShark** [reply]
> In the logs I currently just see floods of incoming MIDI messages of type [SYSTEM_EXCLUSIVE](https://en.wikipedia.org/wiki/MIDI#System_Exclusive_messages) . It's weird your drumkit sends them (especially since its [manual](https://images.static-thomann.de/pics/atg/atgdata/document/manual/c_360221_module_v3_r1_en_online.pdf) states that it is not even capable of sending them) but maybe it was designed to work with a special proprietary software. Those events are not - and can not - be handled by Hydrogen.

Okay, I was plain wrong about this. It's quite plausible for your edrum to send such events and I put some work into supporting such events on Windows as well.

Could you do me a favor and test the [new, patched version](https://ci.appveyor.com/api/buildjobs/w0vlvpafl4eb96ja/artifacts/Hydrogen-1.2.0-win64.exe)?
